### PR TITLE
Created mapping from api call to bills and payments using mock api.

### PR DESCRIPTION
**Pull request by jake-good**
Doesn't directly close an issue, furthers progress for connecting back end but still needs full back end functionality to be tested properly

**Description**

Filled in remainder of the life-cycle method getting bills from API temporarily using a mock API for testing. Created function mapDataToElements that converts the JSON object received from API into a list that is rendered on the page. Currently still renders hard coded transactions as well for reference.

Mob programmed with @pabloSushibar @williamlixu  @vee-christella @jake-good 

**Testing**

Get request has been tested by creating a mock API that returns a JSON object of the same format as the server will once ready. The component successful retrieves object from mock API and renders each respective bill.

Ensured code compiles and does not cause any linting errors. Ensured GUI still appears to function correctly.
**Checklist**

- [x] I have tested my change
- [x] Code builds successfully and all tests pass
- [ ] Docs have been added/updated if needed

### CODE REVIEW

File: split-webapp/split/src/components/TransactionList.js [hunk 1]
```diff
@@ -15,38 +21,67 @@ function ListItemLink(props) {
...
       .then(data => {
         // make a mapping of bills to list items here and render it below
-        console.log(data); // Temporary code (Adjust as required)
+        this.setState({
+          bills: data
+        });
       })
       .catch(err => {
-        alert(err);
+        alert.err(err);
```
Comment (azizul-i) [inline split-webapp/split/src/components/TransactionList.js]: Should ` alert.err(err)` be `alert(err)` ? Just a thought, maybe we should `console.log()` this error for now instead, as having a error pop up might be confusing/strange for a user to see if the promise fails? 

Another alternative is that we make the error message more user friendly, alongside `console.log()` (for debugging purposes)  - This I guess is more a design decision though

File: split-webapp/split/src/components/TransactionList.js [hunk 2]
```diff
@@ -15,38 +21,67 @@ function ListItemLink(props) {
...
+  ));
+}
+
 class TransactionList extends React.Component {
   constructor(props) {
     super(props);
 
     this.state = {
       // If you're using state in component remove the es-lint line below, (else make it a functional component)
       // eslint-disable-next-line react/no-unused-state
```
Comment (azizul-i) [inline split-webapp/split/src/components/TransactionList.js]: We can remove these since the bills[] state is being set within the class 😄

**Review by azizul-i — CHANGES_REQUESTED** [review]
Looks really good!! Just a little query about the catch statement, and whether it should be adjusted to `console.log()`

File: split-webapp/split/src/components/TransactionList.js (hunk 1 above)
Comment (azizul-i) [reply split-webapp/split/src/components/TransactionList.js]: Removing the `alert(err)` would also prevent/fix issue #72

Comment (jake-good) [reply split-webapp/split/src/components/TransactionList.js]: Ok makes sense will fix that 👍

File: split-webapp/split/src/components/TransactionList.js [hunk 3]
```diff
@@ -15,38 +21,67 @@ function ListItemLink(props) {
   return <ListItem button component="a" {...props} />;
 }
 
+function mapDataToElements(bills) {
```
Comment (vee-christella) [inline split-webapp/split/src/components/TransactionList.js]: Little bit of a nit here: Can we change this function name to something like `populateBills()` or something like that? `mapDataToElements()` is a big ambiguous to what data and elements are.

**Review by vee-christella — APPROVED** [review]

**Comment by jake-good** [comment]
Ok I have resolved the comments, should be good to go now :)

**Review by azizul-i — APPROVED** [review]
LGTM ✔️  Great additions!

**Comment by jake-good** [comment]
Also the bills from the Mock API will no be currently appearing on the transaction page when tested as the mock endpoint has reached its daily limit for number of requests.

**Review by williamlixu — APPROVED** [review]
LGTM

**Pull request merged**